app/controllers/whitelist_controller.rb

```diff
@@ -0,0 +1,77 @@
+class WhitelistController < ApplicationController
+
+  # http_basic_authenticate_with name: "cs169", password: ENV['PROJECTSCOPE_PASSWORD']
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  ADMIN_ROLE = "admin"
+  COACH_ROLE = "coach"
+  
+  # GET /whitelist
+  def index
+    if current_user.is_admin?
+       @permitted_users = Authorized_user.all 
+      
+    else
+       flash[:notice] = "You are not authorized to manipulate whitelist."
+       redirect_to projects_url
+    end
+  end
+
+  # GET /whitelist/index
+  def show
+    @readonly = true
+    render :template => 'whitelist/index'
+  end
+  
+  # GET /whitelist/new
+  def new
+    if current_user.is_admin?
+      @authorized_user = Authorized_user.new
+    else
+      flash[:notice] = "You are not authorized to manipulate whitelist."
+      redirect_to projects_url
+    end
+  end
+  
+  # POST /whitelist/
+  def create
+    if current_user.is_admin?
+        email = params[:email]
+        role = params[:role]
+        unless (email =~ VALID_EMAIL_REGEX)
```

Comment:
email format checking can be replaced with this one line at the very top using built in rails validation 
`validates_format_of :email,:with => Devise::email_regexp`